include/swift/AST/DiagnosticsCommon.def

```diff
@@ -66,10 +66,10 @@ ERROR(super_not_in_class_method,none,
 
 ERROR(class_func_not_in_class,none,
       "class methods are only allowed within classes; "
-      "use 'static' to declare a static method", ())
+      "use 'static' to declare a requirement fulfilled by either a class or static method", ())
```

Comment:
"requirement" is the language one uses in the context of protocol members.  I think the confusion in the thread stems from the `class` requirement on protocols not emitting a more refined diagnostic.  You should look into specializing this for exactly that case.